tests/lib/common.py

```diff
@@ -440,8 +440,17 @@ def snmp_mib_get(device, board, iface_ip, mib_name, index, timeout=10, retry=3):
     mib_oid = match.group(1)+'.'+index
     time_out = (timeout*retry)+30
     device.sendline("snmpget -v 2c -c private -t " +str(timeout)+ " -r "+str(retry)+" "+iface_ip+" "+board.mib[mib_name]+"."+str(index))
-    idx = device.expect(['Timeout: No Response from'] + ['iso\.'+mib_oid+'\s+\=\s+\w+\:\s+(.*)\r\n'] + device.prompt, timeout=time_out)
+    idx = device.expect(['Timeout: No Response from'] + ['iso\.'+mib_oid+'\s+\=\s+\S+\:\s+(.*)\r\n'] + device.prompt, timeout=time_out)
     assert idx==1,"Getting the mib %s"% mib_name
     snmp_out = device.match.group(1)
     device.expect(device.prompt)
     return snmp_out
+
+def hex2ipv6(self, hexstr):
+    """
+    Can parse strings in this form:
+    FE 80 00 00 00 00 00 00 3A 43 7D FF FE DC A6 C3
+    """
+    hexstr = hexstr.replace(' ', '').lower()
+    blocks = (''.join(block) for block in zip(*[iter(hexstr)]*4))
+    return ':'.join(str(block) for block in blocks)
```

Comment:
Can we validate this with ipaddress.IPv6Address?

Even if we just resturn a string for now? I'd like to start using ipaddress for all handling of IPs going forward. If this is a new function maybe we can just return:

return ipaddress.IPv6Address(':'.join(str(block) for block in blocks).encode('utf-8'))